udf_js/unzip.sql

```diff
@@ -0,0 +1,53 @@
+/*
+
+Unzips a GZIP string.   
+
+This implementation relies on the zlib.js library (https://github.com/imaya/zlib.js) and 
+the atob function for decoding base64.
+
+*/
+
+
+CREATE TEMP FUNCTION
+  udf_js_unzip (input BYTES)
+  RETURNS STRING
+  LANGUAGE js AS """
+    // converts a byte array to a string
+    function binary2String(byteArray) {
+        return String.fromCharCode.apply(String, byteArray);
+    }
+    
+    // input is in base64, so it needs to be decoded
+    var decodedData = atob(input);
+    var compressedData = decodedData.split('').map(function(e) {
+        return e.charCodeAt(0);
+    });
+    
+    var gunzip = new Zlib.Gunzip(compressedData); 
+    
+    // decompress returns bytes that need to be converted into a string
+    var unzipped = gunzip.decompress();
+    return binary2String(unzipped);
+"""
+OPTIONS (
+  library = "gs://bigquery-etl/gunzip.min.js",
+  library = "gs://bigquery-etl/atob.js"
+);
+
+-- Tests
+
+WITH
+  gzipped AS (
+    SELECT AS VALUE
+      'H4sIAKnBGlwAA6uuBQBDv6ajAgAAAA=='),
```

Comment:
this needs to be decoded so that it is be passed to the udf as `BYTES`
```suggestion
      FROM_BASE64('H4sIAKnBGlwAA6uuBQBDv6ajAgAAAA==')),
```

Reply:
It may be worth having two variants of the function, one that takes in a STRING that's BASE64 encoded bytes, and one that takes in BYTES that aren't BASE64 encoded.

But it's the BYTES version that should be immediately useful for the new raw payload tables we're deploying.

Reply:
I spoke with @scholtzan on zoom and I had misunderstood the interface. I thought that the UDF was defined with a STRING param. I'm surprised that BigQuery accepted a STRING argument.

I'd agree with @relud that I'd like to see this test expressed with `FROM_BASE64('H4sIAKnBGlwAA6uuBQBDv6ajAgAAAA==')` as the input, since that should give us a BYTES field and better approximate real usage.

Reply:
For the record, when I tested this without `FROM_BASE64` it threw an error because the type didn't match udf paramter definition (`input BYTES`)